### PR TITLE
Fix `qml.counts` overflow error for more than 64 qubits

### PR DESCRIPTION
**Pull request by mudit2812**
**Context:**
`qml.counts` was converting samples to integers and using f-strings to format as binary strings, however, it caused wires > 64 to have overflow.

Without the changes in this PR:
```python
import pennylane as qml
import numpy as np

np.random.seed(582)

num_wires = 70
wires = list(range(num_wires))

dev = qml.device("default.clifford", wires=wires, shots=100)

@qml.qnode(dev)
# @qml.defer_measurements
def circuit():
    [qml.PauliX(w) for w in wires]
    return qml.counts()

circuit()
```
```pycon
{'-000000000000000000000000000000000000000000000000000000000000000000001': tensor(100, requires_grad=True)}
```

With the changes:
```python
import pennylane as qml
import numpy as np

np.random.seed(582)

num_wires = 70
wires = list(range(num_wires))

dev = qml.device("default.clifford", wires=wires, shots=100)

@qml.qnode(dev)
# @qml.defer_measurements
def circuit():
    [qml.PauliX(w) for w in wires]
    return qml.counts()

circuit()
```
```pycon
{'1111111111111111111111111111111111111111111111111111111111111111111111': tensor(100, requires_grad=True)}
```

**Description of the Change:**
* Change `qml.counts` to convert directly from list of samples to binary string without first converting to integers.
* Changed `QubitDevice._samples_to_counts` to cast to `int8` instead of `int64`. This is only for casting samples, not converting to ints, as `int8` is more memory efficient.

**Benefits:**
* `qml.counts` now works with arbitrary wires. This is useful for devices like `default.clifford`, which can simulate hundreds of qubits.

**Possible Drawbacks:**

**Related GitHub Issues:**
#5537 , #5513

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -395,6 +395,7 @@
 
 * `qml.counts` no longer returns negative samples when measuring 8 or more wires.
   [(#5544)](https://github.com/PennyLaneAI/pennylane/pull/5544)
+  [(#)]()
```
Comment (mudit2812) [inline doc/releases/changelog-dev.md:398]:
```suggestion
  [(#5556)](https://github.com/PennyLaneAI/pennylane/pull/5556)
```

**Comment by mudit2812** [comment]
[sc-61314]

File: pennylane/measurements/counts.py [hunk 2]
```diff
@@ -307,21 +308,22 @@ def circuit(x):
                 mask = np.logical_not(np.any(mask, axis=tuple(range(1, samples.ndim))))
                 samples = samples[mask, ...]
 
-            # convert to string
-            def convert(x):
-                return f"{x:0{num_wires}b}"
+            def convert(sample):
+                # convert to string
```
Comment (dwierichs) [inline pennylane/measurements/counts.py:312]:
```suggestion
                # convert array of ints to string
```

File: tests/measurements/test_counts.py [hunk 3]
```diff
@@ -145,19 +145,29 @@ def test_counts_with_nan_samples(self):
...
-    @pytest.mark.parametrize("n_wires", [5, 8, 10])
-    @pytest.mark.parametrize("all_outcomes", [True, False])
-    def test_counts_multi_wires_no_overflow(self, n_wires, all_outcomes):
+    @pytest.mark.parametrize("batch_size", [None, 1, 4])
+    @pytest.mark.parametrize("n_wires", [4, 10, 65])
+    @pytest.mark.parametrize("all_outcomes", [False, True])
+    def test_counts_multi_wires_no_overflow(self, n_wires, all_outcomes, batch_size):
         """Test that binary strings for wire samples are not negative due to overflow."""
+        if all_outcomes and n_wires == 65:
+            pytest.skip("Too much memory being used, skipping")
```
Comment (dwierichs) [inline tests/measurements/test_counts.py:154]: We could think about adding a corresponding warning to `qml.counts`, I suppose? :sweat_smile:

**Review by dwierichs — APPROVED** [review]
Looks good to me! Thanks @mudit2812 :tada:

**Review by albi3ro — APPROVED** [review]
Thanks for the explanations.

Documenting for posterity that this PR is more about reverting changes from #5544 that weren't necessary than fixing the bug itself.

**Pull request merged**